### PR TITLE
IISCRUM-1107, improved IBM DB2 monitoring support

### PR DESCRIPTION
**Pull request by bulkware**
I'm no Perl expert so this might not work as I intended. I tested this manually, commenting out the part where the stats file is regenerated. So I have no idea if this works in target platform. I suggest we run this by hand in target environment before deployment (after pull request has gone through, of course).

### CODE REVIEW

File: etc/zabbix/scripts/db2stat.pl [hunk 1]
```diff
@@ -47,12 +46,21 @@
   die "Bad dbname argument";
 }
 
+if ($dbpath =~ /^([-\/\w.]+)$/) {
```
Comment (sapajune) [inline etc/zabbix/scripts/db2stat.pl]: Not sure what characters should be acceptable but I'd assume at least underscore '_' should also be included.

**Comment by kompa3** [comment]
> Add underscore to permitted chars, otherwise looks ok.
> 
> @kompa3 In case the db2stat is already in use, should we also implement migration SQL to update existing items directly in Zabbix?

@Turilas83  what do you think? How big thing is it to manually update all the existing items to provide db2 path parameter as a parameter to the item?

File: etc/zabbix/scripts/db2stat.pl (hunk 1 above)
Comment (bulkware) [reply etc/zabbix/scripts/db2stat.pl]: Isn't underscore included in "\w"?

Comment (sapajune) [reply etc/zabbix/scripts/db2stat.pl]: Certainly looks like it is although I wouldn't personally consider it a "word character". :smile:

**Comment by sapajune** [comment]
> > Add underscore to permitted chars, otherwise looks ok.
> > @kompa3 In case the db2stat is already in use, should we also implement migration SQL to update existing items directly in Zabbix?
> 
> @Turilas83 what do you think? How big thing is it to manually update all the existing items to provide db2 path parameter as a parameter to the item?

This might not be most ideal place for SQL migration since the update cannot be fully automatic because path argument is specific to use case. Just a thought if it might be made easier that way.

File: etc/zabbix/scripts/db2stat.pl (hunk 1 above)
Comment (bulkware) [reply etc/zabbix/scripts/db2stat.pl]: Me neither, I have wondered this several times myself :)

**Pull request merged**